processor/deltatocumulativeprocessor/internal/data/add.go

```diff
@@ -76,6 +106,18 @@ func (dp ExpHistogram) Add(in ExpHistogram) ExpHistogram {
 		hi.SetScale(lo.Scale())
 	}
 
+	// Downscale if an expected number of buckets after the merge is too large.
+	if deltaScale := max(getDeltaScale(dp.Positive(), in.Positive()), getDeltaScale(dp.Negative(), in.Negative())); deltaScale > 0 {
```

Comment:
iiuc, this downscales using the existing algorithm first, possibly exceeding the cap.
we check for that only after the fact, already having allocated more memory than we want, right?
this sounds like it's still vulnerable to DoS attacks

Reply:
Downscaling operation should always produce no more buckets than the original, right?
In particular, `expo.Downscale` never performs allocations (except panicking path), so it can not be a DoS attack vector on memory.